pennylane/ops/op_math/sprod.py

```diff
@@ -145,8 +145,7 @@ def __init__(self, scalar: Union[int, float, complex], base: Operator, do_queue=
         if (base_pauli_rep := getattr(self.base, "_pauli_rep", None)) and (self.batch_size is None):
             scalar = copy(self.scalar)
             if qnp.get_interface(scalar) == "tensorflow" and not scalar.dtype.is_complex:
-                c = qnp.convert_like(1 + 0j, scalar)  # get a complex dtype in the same interface
-                scalar = qnp.cast_like(scalar, c)  # cast scalar to complex dtype
+                scalar = qnp.cast(scalar, "complex128")
```

Comment:
to be consistent with symbolicop change

Reply:
Probably outside the scope of this PR then.  

But we should really come up with a way to be more consistent with precision.  We often treated float32 and complex64 as if they never happen, and convert to float63/ complex128 in a lot of places.